src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,61 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_ALLOWABLE_LENGTH = 6;
+    private static final int MIN_ALLOWABLE_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("User login can`t be null");
+        }
+        loginValidator(user.getLogin());
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("User password can`t be null");
+        }
+        passwordValidator(user.getPassword());
+        if (user.getAge() == null) {
+            throw new InvalidDataException("User age can`t be null");
+        }
+        ageValidator(user.getAge());
+        userIsExist(user);
```

Comment:
You can put all logic of validation in methods:
`if (user.getLogin() == null) {` and `userIsExist` in `loginValidator`
`if (user.getPassword() == null) {` in `passwordValidator` 
and etc.

Reply:
Yes. you are right